### PR TITLE
created entity and stored into core data using Magical Record pod

### PR DESCRIPTION
**Pull request by samsonsunny**

### CODE REVIEW

File: LoginApp/LoginViewController.swift [hunk 1]
```diff
@@ -32,45 +35,69 @@ class LoginViewController: UIViewController {
...
+	}
+	
+	@IBAction func checkLoginCredentials() {
+		
+		if let email = emailField.text, let password = passwordField.text {
 			
-			if let userDatabase:[String: String] = defaults.dictionary(forKey: "userDatabase") as? [String : String]  {
+			let response = validate(email: email, password: password)
+		
+			if !response.0 {
```
Comment (ramesh-lingappan) [inline LoginApp/LoginViewController.swift]: use proper naming for tuples, here 0 and 1 represents nothing, until we dig further inside.

File: LoginApp/LoginViewController.swift [hunk 2]
```diff
@@ -32,45 +35,69 @@ class LoginViewController: UIViewController {
...
-					defaults.set(true, forKey: "isLoggedIn")
-					
-					defaults.synchronize()
-					
-					
+				showAlert(message: response.1)
+				return
+			}
+	
+			if let user: UserInfo = UserInfo.mr_findFirst(byAttribute: "emailId", withValue: email) as? UserInfo
```
Comment (ramesh-lingappan) [inline LoginApp/LoginViewController.swift]: No Password check?

**Review by ramesh-lingappan — CHANGES_REQUESTED** [review]

File: LoginApp/LoginViewController.swift (hunk 1 above)
Comment (samsonsunny) [reply LoginApp/LoginViewController.swift]: Sure I ll use proper naming convention even it is simple logic.

File: LoginApp/LoginViewController.swift (hunk 2 above)
Comment (samsonsunny) [reply LoginApp/LoginViewController.swift]: I forgot to check that. I ll add that too.

File: LoginApp/ChangePasswordViewController.swift [hunk 3]
```diff
@@ -7,56 +7,79 @@
...
-				print("please enter correct old password")
-			}
-		}else{
-			print("please enter both the fields")
+			return
 		}
+		
+		let context	= NSManagedObjectContext.mr_default()
 
+		let emailId = defaults.object(forKey: "loggedInUserEmail") as? String
```
Comment (gupta515) [inline LoginApp/ChangePasswordViewController.swift]: This line, fetching user mailId from user defaults can be in guard statement.

File: LoginApp/ChangePasswordViewController.swift [hunk 4]
```diff
@@ -7,56 +7,79 @@
...
+		guard let user: UserInfo = UserInfo.mr_findFirst(byAttribute: "emailId", withValue: emailId) as? UserInfo else {
+			
+			showAlert(message: "User data is not available in DB")
+			
+			return
+		}
+		
+		guard let password = user.password, password == oldPassword else {
+			
+			showAlert(message: "Please provide old password")
```
Comment (gupta515) [inline LoginApp/ChangePasswordViewController.swift]: Alert should be "Please provide correct old password"

**Pull request merged**